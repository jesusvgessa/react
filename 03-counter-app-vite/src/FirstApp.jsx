import PropTypes from 'prop-types';

export const FirstApp = ({ title, subtitle }) => {

    return (
      <>
        <div data-testid="test-title"> { title } </div>
        {/*<code>{ JSON.stringify(newMessage) }</code>*/}
        <p>{ subtitle }</p>
        <p>{ name }</p>
      </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 0
}