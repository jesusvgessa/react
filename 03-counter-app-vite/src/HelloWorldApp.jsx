// export function App(){
//     // document.createElement...
//     return (<h1>Hola mundo!!!</h1>);
// }

// import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>HelloWorldApp</h1>
  )
}
