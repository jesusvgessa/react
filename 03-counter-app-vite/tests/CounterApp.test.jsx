import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe('Pruebas en <CounterApp />', () => {
   
    test('debe de hacer match con el snapshot', ()=>{

        const initialValue = 10;

        const { container } = render(<CounterApp value={ initialValue }/>)
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el valor inicial de 100', ()=>{

        render( <CounterApp value={100}/>)
        expect( screen.getByText(100) ).toBeTruthy();

    });

    test('debe de incrementar con el boton +1', ()=>{

        render( <CounterApp value={ initialValue }/>)
        fireEvent.click( screen.getByText('+1') );
        screen.debug();
        expect( screen.getByText('11')).toBeTruthy();

    });

});