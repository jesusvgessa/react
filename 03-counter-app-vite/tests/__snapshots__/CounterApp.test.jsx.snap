// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
     
    10
     
  </p>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
